client/src/helpers/fhir/__tests/ServiceRequestBuilder.spec.ts

```diff
@@ -0,0 +1,175 @@
+import { ExtensionUrls } from 'store/urls'
+import { ServiceRequestBuilder } from '../builder/ServiceRequestBuilder'
+import * as mocks from './ServiceRequestBuilder.mocks'
+
+describe('ServiceRequestBuilder - default constructor', () => {
+
+  test('withId - should change id', () => {
+    const builder = new ServiceRequestBuilder()
+    builder.withId('123')
+    const result = builder.build()
+    expect(result.id).toEqual('123')
+  })
+
+  test('withMrn - should change Mrn', () => {
+    const builder = new ServiceRequestBuilder()
+    builder.withMrn('foo', 'bar')
+    const result = builder.build()
+    expect(result.identifier?.[0].value).toEqual('foo')
+    expect(result.identifier?.[0].assigner?.reference).toEqual('Organization/bar')
+  })
+
+  test('withSupervisor - should add /update extension supervisor', () => {
+    const builder = new ServiceRequestBuilder()
+
+    builder.withSupervisor('foo')
+    const added = builder.build()
+    expect(added.extension?.[1].url).toEqual(ExtensionUrls.ResidentSupervisor)
```

Comment:
Question: is extension always listed in a systematic fashion? Said otherwise, is the second element, for instance, [1] always the one we expect?

Plus, maybe you could use a name for the index =>const indexOfYadaYada = 1

Reply:
Extensions don't have specific order.

Reply:
Maybe you create an utils that extract the desired extension based on a criteria?

Reply:
There is one (used in the builder), but for unitary test I don't see the benefit, knowing the index where the extension is supposed to be seems more benefic.